File: solutions/elixir/resistor-color-duo/1/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @resistors %{
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors), do: colors |> Enum.take(2) |> do_value()

  defp do_value(colors, acc \\ 0)
  defp do_value([], acc), do: acc
  defp do_value([color | rest], acc), do: do_value(rest, acc * 10 + @resistors[color])
end

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
BANDS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors):
    band_1, band_2, *_ = colors
    return BANDS.index(band_1) * 10 + BANDS.index(band_2)
